src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaHelpers.cs

```diff
@@ -0,0 +1,188 @@
+            if (userPrefersExpressionBodies == forAnalyzer || (!forAnalyzer && analyzerDisabled))
+            {
+                var expressionBody = GetExpressionBody(declaration);
+                if (expressionBody == null)
+                {
+                    // They don't have an expression body.  See if we could convert the block they 
+                    // have into one.
+
+                    var options = declaration.SyntaxTree.Options;
+                    var conversionPreference = forAnalyzer ? preference : ExpressionBodyPreference.WhenPossible;
+
+                    return TryConvertToExpressionBody(declaration, options, conversionPreference, out _, out _);
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryConvertToExpressionBody(
+            LambdaExpressionSyntax declaration,
+            ParseOptions options, ExpressionBodyPreference conversionPreference, 
+            out ExpressionSyntax expressionWhenOnSingleLine, 
+            out SyntaxToken semicolonWhenOnSingleLine)
+        {
+            return TryConvertToExpressionBodyWorker(
+                declaration, options, conversionPreference,
+                out expressionWhenOnSingleLine, out semicolonWhenOnSingleLine);
+        }
+
+        private static bool TryConvertToExpressionBodyWorker(
+            LambdaExpressionSyntax declaration, ParseOptions options, ExpressionBodyPreference conversionPreference,
+            out ExpressionSyntax expressionWhenOnSingleLine, out SyntaxToken semicolonWhenOnSingleLine)
+        {
+            var body = declaration.Body as BlockSyntax;
+
+            return body.TryConvertToExpressionBody(
+                declaration.Kind(), options, conversionPreference,
+                out expressionWhenOnSingleLine, out semicolonWhenOnSingleLine);
+        }
+
+        public static (bool canOffer, bool fixesError) CanOfferUseBlockBody(
+            SemanticModel semanticModel, OptionSet optionSet,
+            LambdaExpressionSyntax declaration, bool forAnalyzer,
+            CancellationToken cancellationToken)
+        {
+            var currentOptionValue = optionSet.GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedLambdaExpressions);
+            var preference = currentOptionValue.Value;
+            var userPrefersBlockBodies = preference == ExpressionBodyPreference.Never;
+            var analyzerDisabled = currentOptionValue.Notification.Severity == ReportDiagnostic.Suppress;
+
+            var expressionBodyOpt = GetExpressionBody(declaration);
+            if (expressionBodyOpt == null)
+            {
+                return (canOffer: false, fixesError: false);
+            }
+
+            var lambdaType = semanticModel.GetTypeInfo(declaration, cancellationToken).ConvertedType as INamedTypeSymbol;
+            if (lambdaType == null || lambdaType.DelegateInvokeMethod == null)
```

Comment:
Same suggestion, I *think* `GetSymbolInfo` should return a method symbol for the lambda

Reply:
Also, why is this necessary?

Reply:
yes.  in error situations. because we need the method symbol to determine what sort of block to create, i was not providing the fix here.

Reply:
seems reasonable

Reply:
This is one of the areas where it would be appropriate to offer a refactoring but with an analyzer it may be good to err on the side of caution